documentation/disaster-recovery-testing.md

```diff
@@ -0,0 +1,176 @@
+
+Confirm the app is working and can see the restored data
+
+
+## Scenario 2. Loss of data
+
+In the case of data loss, we need to recover the data as soon as possible in order to resume normal service.
+
+The application's database is an Azure flexible postgres instance. This provides a point-in-time restore (PTR) ability with the resolution of 1 second, available between 5min and 7days. PTR allows you to restore the source server to a point-in-time on a new copy of the server, not to the source server itself.
+
+The objectives of this test are:
+- Create a separate new postgres database instance
+- Restore data from the current live database to the new postgres database instance from a particular point in time
+- data can then be updated into the source database from the new PTR server
+    - do we want to test this as part of the DR test?
+
+### Delete data from the postgres database instance
+
+Make a note of the time this step is being started as the restore point must be before you delete any data
+
+- Delete a table manually
+    - connect via konduit and delete the table, or can this be done within the UI?
+- confirm it's deleted?
+
+If deleting any other resources (scenario 3) then this should be done at this point too.
+
+### Recreate any deleted resources
+
+Only required if any resources have been deleted (scenario 3)
+
+As the maintenance page has been implemented, you will need to
+- create a branch from main
+- update the env tfvars.json as per https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md#configure-terraform-to-keep-deploying-the-application
+- push the branch to github
+- run the deploy workflow using your branch for the env being tested
+
+### Restore postgres database instance
+
+Run the point in time restore workflow using a time before the data was deleted
+- Important: You should convert the time to UTC before actually using it. When you record the time, note what timezone you are using. Especially during BST (British Summer Time).
+
+### Restore data into the source server
+
+Connect to the copy db via konduit and use pg_dump or similar
+- restore into the existing db using konduit
+- create a workflow to do this?
+
+### Validate app
+
+Confirm the app is working and can see the restored data
+
+### Tidy up
+
+Once the database has been successfully restored, the database copy instance should be deleted.
+
+
+## Post scenario steps
+
+## Disable maintenance mode
+
+Run the disable maintenance workflow for the service and environment affected
+
+## Enable statuscake monitoring
+
+Enable and disabled or paused statuscake monitoring
+
+## Unfreeze pipeline
+
+Alert developers that merge to main is allowed.
+- In github settings, update the Branch protection rules and set required PR approvers back to 1
+
+## Documentation requirements
+
+- Complete the DR testing document
+    - record of the steps run, time taken, and any issues
+- Save the document here (tba)
```

Comment:
Here or in each service repo?

Reply:
A directory in sharepoint. The DR test document would be a separate template document.